### ISSUE TITLE
[Bug] Inconsistent behavior when comparing property values with a string value

### ISSUE THREAD

**Issue by zhengyingying** [apache/incubator-hugegraph#2164]
### Bug Type (问题类型)

gremlin (结果不合预期)

### Before submit

- [X] 我已经确认现有的 [Issues](https://github.com/apache/hugegraph/issues) 与 [FAQ](https://hugegraph.apache.org/docs/guides/faq/) 中没有相同 / 重复问题 (I have confirmed and searched that there are no similar problems in the historical issue and documents)

### Environment (环境信息)

- Server Version: 1.0.0 (Apache Release Version)
- Backend: RocksDB x nodes, HDD or SSD 
- OS: xx CPUs, xx G RAM, Ubuntu 2x.x / CentOS 7.x 
- Data Size:  xx vertices, xx edges 

### Expected & Actual behavior (期望与实际表现)

I have noticed that HugeGraph does not support `gte(STRING)` from the previous issues. However, I can get the expected result when I remove `HugeGraphStepStrategy`.  

Does `HugeGraphStepStrategy` add some restrictions? I think they should behave consistently.

```
gremlin> :> hugegraph.traversal().V().has('vp0', not(gte('10')))
Can't compare between 1(String) and 10(String)
 
gremlin> :> hugegraph.traversal().withoutStrategies(HugeGraphStepStrategy).V().has('vp0', not(gte('10')))
==>[id:700015037397336064,label:vl0,type:vertex,properties:[vp0:1]]
```

**Comment by javeme** [comment]
Yes you are right, now the HugeGraphStepStrategy just supports numbers comparison.

**Comment by zhengyingying** [comment]
Thank you for your reply. Looking forward to the new feature in the future.
